src/tmpfiles/tmpfiles.c

```diff
@@ -1614,7 +1615,11 @@ static int create_directory_or_subvolume(const char *path, mode_t mode, bool sub
                 return pfd;
 
         if (subvol) {
-                if (btrfs_is_subvol(empty_to_root(arg_root)) <= 0)
+                if ((getenv_bool("SYSTEMD_TMPFILES_FORCE_SUBVOL") > 0 && btrfs_is_filesystem(pfd) > 0) ||
+                    btrfs_is_subvol(empty_to_root(arg_root)) > 0) {
+                        RUN_WITH_UMASK((~mode) & 0777)
+                                r = btrfs_subvol_make_fd(pfd, basename(path));
+                } else
```

Comment:
why not this:

```c
r = getenv_bool("SYSTEMD_TMPFILES_FORCE_SUBVOL");
if (r < 0) {
        log_warning_errno(r, "Cannot parse value of $SYSTEMD_TMPFILES_FORCE_SUBVOL, ignoring.");
        r = btrfs_is_subvol(empty_to_root(arg_root)) > 0;
} 
if (!r)
        subvol = false;
else {
        RUN_WITH_UMASK(…)
        …
}
```

i.e. stay close to the old behaviour, just replace the `btrfs_is_subvol(empty_to_root(arg_root)) <= 0` check with the getenv check if the env var is set

Reply:
Alright I can do that. Where should I check for another FS then?

Something like this?
```
r = getenv_bool("SYSTEMD_TMPFILES_FORCE_SUBVOL");
if (r < 0) {
        log_warning_errno(r, "Cannot parse value of $SYSTEMD_TMPFILES_FORCE_SUBVOL, ignoring.");
        r = btrfs_is_subvol(empty_to_root(arg_root)) > 0;
} else if (r > 0) {
        r = btrfs_is_filesystem(pfd) > 0;
}
if (!r)
        subvol = false;
else {
        RUN_WITH_UMASK(…)
        …
}
```

Reply:
hmm? btrfs_subvol_make_fd() will fail with ENOTTY on other file systems, and there's a fallback in place for that, so not sure what else you need?